edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeAuthenticator.cs

```diff
@@ -103,7 +106,7 @@ async Task<(bool isAuthenticated, bool serviceIdentityFound)> AuthenticateWithSe
         {
             Option<string> authChain = await this.deviceScopeIdentitiesCache.GetAuthChain(serviceIdentityId);
 
-            if (!authChain.HasValue)
+            if (this.nestedEdgeEnabled && !authChain.HasValue)
```

Comment:
Previously it will return as long as authChain doesn't have value.  why only return here if nested edge is enabled?

Reply:
This check only applies in nested edge scenarios, previously it assumed nested edge is always enabled.